lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -1838,8 +1838,9 @@ static ObjCSelector inferObjCName(ValueDecl *decl) {
         auto diag = decl->diagnose(diag::objc_ambiguous_inference_candidate,
                                    req, proto, *req->getObjCRuntimeName());
         fixDeclarationObjCName(diag, decl,
-                               decl->getObjCRuntimeName(/*skipIsObjC=*/true),
-                               req->getObjCRuntimeName());
+                               decl->getObjCRuntimeName(/*skipIsObjC=*/true)
+                                 .value_or(ObjCSelector()),
+                               req->getObjCRuntimeName().value());
```

Comment:
It's hard for me to tell - is there a reason we can be certain that `req->getObjCRuntimeName()` here and in the `if` above has a value?

Reply:
Good question. I *think* the answer is that (a) `findWitnessedObjCRequirements()` can only return requirements of an `isObjC()` protocol, and (b) an `isObjC()` protocol's requirements are always `isObjC()`, so `getObjCRuntimeName()` will always be able to return a name.